src/main/java/com/hackathon03/codeguru/utils/InEfficientMapEntryIteration.java

```diff
@@ -0,0 +1,20 @@
+package com.hackathon03.codeguru.utils;
+
+import java.util.Map;
+
+public class InEfficientMapEntryIteration {
+    public void iterateOnKeySetUsingValuesNoncompliant(Map<String, String> map) {
+        // Noncompliant: iterate the set of keys and get the value of each key from the map.
+        for (String name : map.keySet())
+            System.out.println("Value: " + map.get(name));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.